src/Symfony/Component/Translation/Translator.php

```diff
@@ -372,6 +372,39 @@ private function doLoadCatalogue($locale)
                 $this->catalogues[$locale]->addCatalogue($this->loaders[$resource[0]]->load($resource[1], $locale, $resource[2]));
             }
         }
+
+        $this->resolveCatalogueMessages($this->catalogues[$locale]);
+    }
+
+    private function resolveCatalogueMessages($catalogue)
+    {
+        $expr = '{(.*?)}';
+        foreach ($catalogue->all() as $domain => $messages) {
+            preg_match_all('/'.$expr.'/', implode($messages), $matches);
+            $parameters = [];
+            foreach ($matches[1] as $key => $id) {
+                if (isset($messages[$id])) {
+                    $parameters[$matches[0][$key]] = $messages[$id];
+                }
+            }
+
+            while (preg_match_all('/'.$expr.'/', implode($parameters))) {
```

Comment:
Why not putting the delimiters directly into the `$expr` variable? You're duplicating the delimiters both at line 383 and 391.